yggdrasilffi/src/lib.rs

```diff
@@ -41,17 +42,32 @@ impl<T> From<Result<Option<T>, FFIError>> for Response<T> {
             Err(e) => Response {
                 status_code: ResponseCode::Error,
                 value: None,
-                error_message: Some(format!("{:?}", e)),
+                error_message: Some(e.to_string()),
             },
         }
     }
 }
 
-#[derive(Debug)]
 enum FFIError {
     Utf8Error,
-    InvalidJson,
     NullError,
+    InvalidJson(String),
+    PartialUpdate(Vec<EvalWarning>),
+}
+
+impl Display for FFIError {
+    fn fmt(&self, f: &mut Formatter) -> fmt::Result {
+        match self {
+            FFIError::Utf8Error => write!(f, "Detected a non UTF-8 string in the input, this is a serious issue and you should report his as a bug."),
```

Comment:
```suggestion
            FFIError::Utf8Error => write!(f, "Detected a non UTF-8 string in the input, this is a serious issue and you should report this as a bug."),
```

Reply:
:sweat_smile: Sometimes I English well but sometimes, no

Thank you